packages/fxa-settings/src/pages/Signin/SigninTotpCode/index.tsx

```diff
@@ -51,20 +61,35 @@ const SigninTotpCode = ({
     submitButtonText: 'Confirm',
   };
 
-  const onSubmit = () => {
-    try {
-      // Check authentication code
-      // logViewEvent('flow', `${viewName}.submit`, ENTRYPOINT_REACT);
-      // Check if isForcePasswordChange
-    } catch (e) {
-      // TODO: error handling, error message confirmation
-      // this should probably use auth-errors and message should be displayed in tooltip or banner
+  const onSubmit = async (code: string) => {
+    const result = await submitTotpCode(code);
+
+    setGeneralError('');
+    setCodeErrorMessage('');
+
+    if (result === true) {
+      setSuccess(true);
+      handleNavigation();
+    } else if (result === false) {
```

Comment:
In other container components, we handle the error by returning a particular error shape with at least `errno, message, ftlId`. Should we do that here for consistency? Relates to Valerie's comment here: https://github.com/mozilla/fxa/pull/16389/files#r1484881811

Reply:
I was on the fence on this. Part of my thinking was that error handling should be contained by the container, but I think the error is tied closely enough to the UI state that passing it is probably better.

Reply:
Yeah I can see it both ways too... If you check `cachedSigninHandler` in Signin Container, you can see sometimes that we need to manipulate what kind of data and error is returned. I kind of like the idea of always returning a particular `error` shape from container components despite whether GQL or auth-client is used, but our patterns are kind of in flight so happy to see alternatives.